### PR TITLE
[Merged by Bors] - chore(Order): Make more arguments explicit

### PR DESCRIPTION
**Pull request by YaelDillies**
Those lemmas have historically been very annoying to use in `rw` since all their arguments were implicit. One too many people complained about it on Zulip, so I'm changing them.

Downstream code broken by this change can fix it by adding appropriately many `_`s.

Also marks `CauSeq.ext` `@[ext]`.

## `Order.BoundedOrder`

* `top_sup_eq`
* `sup_top_eq`
* `bot_sup_eq`
* `sup_bot_eq`
* `top_inf_eq`
* `inf_top_eq`
* `bot_inf_eq`
* `inf_bot_eq`

## `Order.Lattice`

* `sup_idem`
* `sup_comm`
* `sup_assoc`
* `sup_left_idem`
* `sup_right_idem`
* `inf_idem`
* `inf_comm`
* `inf_assoc`
* `inf_left_idem`
* `inf_right_idem`
* `sup_inf_left`
* `sup_inf_right`
* `inf_sup_left`
* `inf_sup_right`

## `Order.MinMax`

* `max_min_distrib_left`
* `max_min_distrib_right`
* `min_max_distrib_left`
* `min_max_distrib_right`

---

- [ ] depends on: #11144

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Can you list the lemmas changes in the description? Just their names is fine, you don't need to give the full diff of the signatures.

File: Mathlib/Order/Lattice.lean [hunk 1]
```diff
@@ -218,38 +218,38 @@ theorem sup_le_sup_right (h₁ : a ≤ b) (c) : a ⊔ c ≤ b ⊔ c :=
...
-instance : Std.Associative (α := α)  (· ⊔ ·) := ⟨@sup_assoc _ _⟩
+instance : Std.Associative (α := α)  (· ⊔ ·) := ⟨sup_assoc⟩
 
 theorem sup_left_right_swap (a b c : α) : a ⊔ b ⊔ c = c ⊔ b ⊔ a := by
-  rw [sup_comm, @sup_comm _ _ a, sup_assoc]
+  rw [sup_comm, sup_comm a, sup_assoc]
 #align sup_left_right_swap sup_left_right_swap
 
-theorem sup_left_idem : a ⊔ (a ⊔ b) = a ⊔ b := by simp
+lemma sup_left_idem (a b : α) : a ⊔ (a ⊔ b) = a ⊔ b := by simp
```
Comment (eric-wieser) [inline Mathlib/Order/Lattice.lean:241]: I'd prefer not to adjust `lemma`/`theorem` as part of this diff, it just makes it noisy.

File: Mathlib/Order/MinMax.lean [hunk 2]
```diff
@@ -119,20 +119,20 @@ theorem min_lt_of_right_lt (h : b < c) : min a b < c :=
...
 
-theorem min_max_distrib_left : min a (max b c) = max (min a b) (min a c) :=
-  inf_sup_left
+lemma min_max_distrib_left(a b c : α) : min a (max b c) = max (min a b) (min a c) :=
+  inf_sup_left _ _ _
 #align min_max_distrib_left min_max_distrib_left
 
-theorem min_max_distrib_right : min (max a b) c = max (min a c) (min b c) :=
-  inf_sup_right
+lemma min_max_distrib_right(a b c : α) : min (max a b) c = max (min a c) (min b c) :=
```
Comment (eric-wieser) [inline Mathlib/Order/MinMax.lean:134]: missing space:
```suggestion
ttheorem min_max_distrib_right (a b c : α) : min (max a b) c = max (min a c) (min b c) :=
```

Comment (eric-wieser) [reply Mathlib/Order/MinMax.lean:134]: (and above)

File: Mathlib/RingTheory/Ideal/Operations.lean [hunk 3]
```diff
@@ -1092,20 +1090,9 @@ variable {R}
...
-              rw [pow_succ]
-              exact radical_mul _ _
-            _ = radical I ⊓ radical I := by rw [ih H]
-            _ = radical I := inf_idem
-            )
-        fun H => H ▸ (pow_one I).symm ▸ rfl)
-    H
+lemma radical_pow : ∀ n, 0 < n → radical (I ^ n) = radical I
+  | 1, _ => by simp
+  | n + 2, _ => by rw [pow_succ, radical_mul, radical_pow _ n.succ_pos, inf_idem]
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Operations.lean:1095]: This golf probably doesn't belong hidden in this PR.

Comment (YaelDillies) [reply Mathlib/RingTheory/Ideal/Operations.lean:1095]: Sure, but I had to fix the `inf_idem` call in the middle of the proof

File: Mathlib/Order/Lattice.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Order/Lattice.lean:241]: This is kind of a pain to undo

File: Mathlib/RingTheory/Ideal/Operations.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/RingTheory/Ideal/Operations.lean:1095]: #11144

**Review by eric-wieser — APPROVED** [review]
bors d+

I reverted the `theorem`/`lemma` switch myself

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**